CMakeLists.txt

```diff
@@ -931,9 +931,12 @@ if(HAVE_OPENVX)
 endif()
 
 if(WITH_KLEIDICV)
-  ocv_debug_message(STATUS "Enable KleidiCV acceleration")
+  ocv_debug_message(STATUS "Try to enable KleidiCV acceleration")
   if(NOT ";${OpenCV_HAL};" MATCHES ";kleidicv;")
-    set(OpenCV_HAL "kleidicv;${OpenCV_HAL}")
+    add_subdirectory(3rdparty/kleidicv)
```

Comment:
This skips `add_subdirectory()` call if user specifies `cmake -DOpenCV_HAL=kleidicv ...`.

Reply:
What is the problem to follow carotene design?